.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,102 @@
+          )" >> $GITHUB_OUTPUT
+          echo "short_cid=${CLIENT_ID:0:8}" >> $GITHUB_OUTPUT
+      - name: Upload sample data
+        id: upload-sample-data
+        env:
+          TF_ENV: ${{ steps.set-environment.outputs.tf_env }}
+          SHORT_CID: ${{ steps.set-environment.outputs.short_cid }}
+        run: |
+          TIMESTAMP=$(date "+%Y-%m-%dT%H:%M:%S")
+          echo "timestamp=$TIMESTAMP" >> $GITHUB_OUTPUT
+          az storage blob upload --account-name phdi${TF_ENV}phi${SHORT_CID} --container-name source-data --name vxu/VXU-V04-01_success_single_$TIMESTAMP.hl7 --file sample-data/VXU-V04-01_success_single.hl7
+      - name: Check pipeline run
+        env:
+          TF_ENV: ${{ steps.set-environment.outputs.tf_env }}
+          SHORT_CID: ${{ steps.set-environment.outputs.short_cid }}
+          TIMESTAMP: ${{ steps.upload-sample-data.outputs.timestamp }}
+        run: |
+          az extension add --name datafactory --upgrade
+          START_DATE=$(date --date="1 day ago" "+%Y-%m-%dT%H:%M:%S")
+          END_DATE=$(date --date="1 day" "+%Y-%m-%dT%H:%M:%S")
+          SOURCE_FILE="source-data/vxu/VXU-V04-01_success_single_$TIMESTAMP.hl7"
+          CHECK_COUNT=0
+
+          check_pipeline_run_count() {
+            az datafactory pipeline-run query-by-factory --factory-name "phdi-$TF_ENV-data-factory-$SHORT_CID" --filters operand="PipelineName" operator="Equals" values="phdi-$TF_ENV-ingestion" --filters operand="Status" operator="Equals" values="$1" --last-updated-after "$TIMESTAMP" --last-updated-before "$END_DATE" --query "value" | jq --arg SOURCE_FILE "$SOURCE_FILE" 'map(select(.parameters.filename == $SOURCE_FILE)) | length'
+          }
+
+          check_pipeline_success_count() {
+            check_pipeline_run_count "Succeeded"
+          }
+
+          check_pipeline_failure_count() {
+            check_pipeline_run_count "Failed"
+          }
+
+          while [[ "$(check_pipeline_success_count)" -lt 1 ]]; do
+            if [[ "$CHECK_COUNT" -gt 60 || "$(check_pipeline_failure_count)" -gt 0 ]]; then
+              echo "Pipeline run failed"
+              exit 1
+            fi
+            echo "Waiting for pipeline to complete run..."
+            sleep 10
+            CHECK_COUNT=$((CHECK_COUNT+1))
+          done
+
+          echo "Pipeline run succeeded!"
+
+      - name: Query FHIR server
+        env:
+          TF_ENV: ${{ steps.set-environment.outputs.tf_env }}
+          SHORT_CID: ${{ steps.set-environment.outputs.short_cid }}
+        run: |
+          TOKEN=$(az account get-access-token --resource=https://${TF_ENV}fhir${SHORT_CID}.azurehealthcareapis.com --query accessToken --output tsv)
+          RESPONSE=$(curl -X GET --header "Authorization: Bearer $TOKEN" "https://${TF_ENV}fhir${SHORT_CID}.azurehealthcareapis.com/Patient?family=EVERYMAN&given=ADAM")
+          echo $RESPONSE | jq -e '.entry[0].resource.name[0].family == "EVERYMAN" and .entry[0].resource.name[0].given[0] == "ADAM"'
```

Comment:
What happens here when the end to end tests runs after the first time in a given environment? Since we leave the data in the server after the test we would still expect it to be there when the query is run again.

Reply:
Good point. Is there a field like `lastUpdated` or something where we can check to see if the record was recently updated?

Reply:
`_lastUpdated` https://build.fhir.org/search.html#_lastUpdated

Reply:
There is also a `_since` options, but I think it is only for resource creation date.